test/dummy_test.go

```diff
@@ -0,0 +1,134 @@
+package test
+
+import (
+	"context"
+	"testing"
+	"time"
+
+	"github.com/stretchr/testify/assert"
+
+	"github.com/rollkit/go-sequencing"
+)
+
+func TestSubmitRollupTransaction(t *testing.T) {
+	sequencer := NewDummySequencer()
+
+	// Define a test rollup ID and transaction
+	rollupId := []byte("test_rollup_id")
+	tx := []byte("test_transaction")
+
+	// Submit a transaction
+	req := sequencing.SubmitRollupTransactionRequest{
+		RollupId: rollupId,
+		Tx:       tx,
+	}
+	resp, err := sequencer.SubmitRollupTransaction(context.Background(), req)
+
+	// Assert no error
+	assert.NoError(t, err)
+	// Assert the transaction was successfully added to the queue
+	assert.NotNil(t, resp)
+	assert.Equal(t, rollupId, sequencer.RollupId)
+}
+
+func TestSubmitRollupTransaction_RollupIdMismatch(t *testing.T) {
```

Comment:
Building on my other comment, I don't think `SubmitRollupTransaction` should set the rollupId for the sequencer. 
If the rollupId should be unique and verifiable, then I think it should be explicitly set, not implicitly set. 

So maybe a config of the sequencer. 

In production, then this could be a CLI argument or in the toml file. 

In testing, the `NewDummySequencer` should set a unique rollupId. The test can access it to test using that vs a different rollupId.

Reply:
yes, we don't need to store rollup id. I have removed it.

Reply:
i correct myself. we need this to prevent multiple rollup ids using same sequencer middleware. but moved the rollup id to NewDummySequencer constructor now.